update vector search docs (#18779)
pingcap/docs: vector-search-overview.md

---
title: Vector Search Overview
summary: Learn about Vector Search in TiDB Cloud. This feature provides an advanced search solution for performing semantic similarity searches across various data types, including documents, images, audio, and video.
---

# Vector Search Overview

TiDB Vector Search provides an advanced search solution for performing semantic similarity searches across various data types, including documents, images, audio, and video. This feature enables developers to easily build scalable applications with generative artificial intelligence (AI) capabilities using familiar MySQL skills.

<CustomContent platform="tidb">

> **Warning:**
>

> > The vector search feature is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer: ```suggestion
> The vector search feature is experimental and may be not fully functional. It is not recommended that you use it in the production environment. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

Remove "This feature might be changed or removed without prior notice." as this feature is experimental because of stability issues, not production decision issues.

Reviewer 2: Updated via https://github.com/pingcap/docs/pull/18779/commits/a301f355322c72dcf8ca7453e4e0841b0a142e0e.

</CustomContent>

<CustomContent platform="tidb-cloud">

> **Warning:**
>

> > The vector search feature is in beta. It might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer: ```suggestion
> The vector search feature is in beta and some behaviors may change in future versions. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

Even when the search terms do not exactly match the content in the database, vector search can still provide results that align with the user's intent by analyzing the semantics of the data.

For example, a full-text search for "a swimming animal" only returns results containing these exact keywords. In contrast, vector search can return results for other swimming animals, such as fish or ducks, even if these results do not contain the exact keywords.

### Vector embedding

A vector embedding, also known as an embedding, is a sequence of numbers that represents real-world objects in a high-dimensional space. It captures the meaning and context of unstructured data, such as documents, images, audio, and videos.

Vector embeddings are essential in machine learning and serve as the foundation for semantic similarity searches.

> TiDB introduces [Vector data types](/vector-search-data-types.md) designed to optimize the storage and retrieval of vector embeddings, enhancing their use in AI applications. You can store vector embeddings in TiDB and perform vector search queries to find the most relevant data using these data types.

Author: ```suggestion
TiDB introduces [Vector data types](/vector-search-data-types.md) and [Vector search indexes](/vector-search-index) designed to optimize the storage and retrieval of vector embeddings, enhancing their use in AI applications. You can store vector embeddings in TiDB and perform vector search queries to find the most relevant data using these data types.
```
